pkg/operator/ceph/config/keyring/store.go

```diff
@@ -90,6 +92,10 @@ func (k *SecretStore) CreateOrUpdate(resourceName string, keyring string) error
 		},
 		Type: k8sutil.RookType,
 	}
+	if secret.Name == "rook-ceph-admin-keyring" {
```

Comment:
Instead of checking for a specific secret name, how about if we add a new helper method such as `CreateOrUpdateWithClusterAnnotations()`, which would only be called for this secret. This `CreateOrUpdate()` method could just be factored out internally to avoid code duplication.

Also with this approach, the new method could take the clusterSpec as an arg, and hopefully no need for it on the ClusterInfo struct anymore.

Reply:
And how about a unit test for this?

Reply:
Actually, we want the secret `rook-ceph-mon`, correct? I believe that secret will also cover this admin keyring secret.

Reply:
in the issue, it was mentioned both secrets but I agree `rook-ceph-mon` will cover but I went ahead with issue request.